source/js/buyers-guide/template-js-handler/product-page-comment-handler.js

```diff
@@ -4,34 +4,48 @@
  * we have this file which will push a datalayer event whenever a user submits a comment.
  */
 export default () => {
+  // Commento's parent div, in which it loads all of its content through JS.
   const commentoContainer = document.querySelector(
     "#view-product-page #commento"
   );
-  const submitButton = document.querySelector("#commento-submit-button-root");
 
-  if (commentoContainer && submitButton) {
-    submitButton.addEventListener(`click`, () => {
-      commentSubmitted();
+  const commentListUpdateHandler = (mutations) => {
+    let commentList = document.querySelector("#commento-main-area");
```

Comment:
The reason for this line is that whenever a user logs in/out, the "#commento-main-area" div gets deleted and re-rendered.
 So if we were to define this once somewhere else in the code and then reference it here, and the user had to log in before submitting a comment (which I think is going to be the case for most users), this object would no longer exist.